### ISSUE TITLE
Corrupted log in tensorboard

### ISSUE THREAD

**Issue by dchatterjee172** [PyTorchLightning/pytorch-lightning#3446]
<img width="865" alt="Screen Shot 2020-09-10 at 4 52 37 PM" src="https://user-images.githubusercontent.com/10482742/92722994-0a2b5280-f386-11ea-9d28-a1418f326f44.png">

```
        result = pl.TrainResult(minimize=loss)
        if compute_metrics:
            result.log("train_loss", loss, sync_dist=True)
            result.log("train_bleu", bleu, sync_dist=False)
            result.log("train_em", em, sync_dist=False)

```
bleu and em are metrics defined by subclassing `pl.metrics.metric.TensorMetric`
```
class ExactMatch(pl.metrics.metric.TensorMetric):
```
At epoch 0 everything works fine, but when transitioning from epoch 0 to 1 after completing validation in epoch 0, something happens, and the loss and other metrics reduce suddenly.

Now the blue graph is resumed training run of the same model (orange line).

I am using DDP.

`compute_metrics` is defined by 
```
        compute_metrics = (
            (batch_idx + 1) % self.trainer.row_log_interval == 0
            or self.trainer.should_stop
            or self.trainer.fast_dev_run
        )
```
I did not want to compute BLEU and EM for all batch in the training. The bleu and em only are computed when it is time to log. If I do not do this and remove `if compute_metrics:`, it works fine. As demonstrated below.
![image](https://user-images.githubusercontent.com/10482742/92728837-110a9300-f38f-11ea-95f9-3fb5628c9b9d.png)

As per as I understand `result.log` does not do any accumulation. Then why is this happening?

#### What's your environment?

 - OS: [e.g. iOS, Linux, Win] Linux
 - Packaging [e.g. pip, conda] pip
 - Version [e.g. 0.5.2.1] 0.9.1
